app/scripts/metamask-controller.js

```diff
@@ -3526,6 +3528,9 @@ export default class MetamaskController extends EventEmitter {
       });
     };
     this.on('startUISync', startUISync);
+    if (this.startUISync) {
+      this.emit('startUISync');
+    }
```

Comment:
In case event listeners were not created when `this.emit('startUISync')` was called, code above will take care to send event to UI.

Reply:
good catch!

Reply:
Nit: this might make more sense reformatted as something like

```
if (this.startUISync) {
  startUISync()
} else {
  this.on('startUISync', startUISync);
}
```

e.g. call sync now, otherwise schedule for later. As we don't need the event listener at all if the UI sync is already prepared.